src/lib_json/json_reader.cpp

```diff
@@ -1378,30 +1378,39 @@ bool OurReader::readCppStyleComment() {
   }
   return true;
 }
-
-bool OurReader::readNumber(bool checkInf) {
+// if checkNeg is true, read negative number
+bool OurReader::readNumber(bool checkNeg) {
   const char* p = current_;
-  if (checkInf && p != end_ && *p == 'I') {
-    current_ = ++p;
-    return false;
+  char c = '0'; // stopgap for already consumed character  
+  if(checkNeg){
```

Comment:
Nit: space around brackets. Do we use clang format or anything? We definitely should.